src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -158,6 +156,34 @@ private async Task<IActionResult> GetResultAsync(HttpContext context, IFunctionE
             return new OkResult();
         }
 
+        private void PopulateRouteData(HttpContext context)
+        {
+            // Add route data to request info
+            // TODO: Keeping this here for now as other code depend on this property, but this can be done in the HTTP binding.
```

Comment:
This is a TODO that was here before, but all of this logic can be moved to the HTTP binding, eliminating the need to pass the state around in the items.